martinsluters/lagom-blocks
# Add Gutenberg block that is a wrapper of WP core Heading block and WooCommerce core Newest Products block

# Purpose
There is a lack of inherent justification for the creation of this block, as it neither alters the functionality nor the style of the core blocks. The wrapper block itself does not introduce any distinctive styling. It would be more pragmatic to adhere to the native offerings of the Gutenberg editor to achieve equivalent outcomes. Specifically, incorporating Heading and Newest Products blocks and adjusting their settings to the desired appearance is a more streamlined approach. These two blocks can then be encapsulated within a Group block, or even a Columns block if necessary. If the intended result is to be replicated across multiple pages, the utilization of a block pattern can effectively enhance efficiency through the DRY principle.

The current Pull Request (PR) is formulated solely as a proof of concept, demonstrating the feasibility of creating a block with <InnerBlock>, encompassing permitted blocks and a pre-defined template of blocks with their respective attributes.

**Block's name is `lagom-blocks/heading-products` and title is "Heading & Products".
It can be found under WooCommerce category of blocks.**

## Testing

**PHP Integration tests:**
To run phpunit run `npm run test:phpunit` command in terminal.

**E2E tests:**
To run e2e tests run either `npm run test:e2e` or `npm run test:e2e:debug` commands in terminal.

**Manual tests**
To manually test the block:
1. Log in http://localhost:8888/wp-admin/ . Refer to [wp-env ](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-env/) documentation for the credentials.
2. Run command `npm run seed-fixtures` to set up some test WooCommerce products. (Optional)
3. Go to http://localhost:8888/wp-admin/post-new.php?post_type=page
4. Look for block Heading & Products block
5. Observe the block is added to the editor
6. Publish page
7. Go to the published page's public URL and observe the contents of the block.

**Screencast of manual testing**

https://github.com/martinsluters/lagom-blocks/assets/18058037/f0b785c6-5d01-4322-9181-87c56e1fcd18


> ⚠️ When testing the block, this PR assumes that Docker is running and `wp-env` is running. All the dependencies PHP Composer and npm are installed. This PR assumes that a developer knows how to start Docker, initiate `wp-env` against this project and install dependencies.
